### ISSUE TITLE
Add JRE 15 alpine tag

### ISSUE THREAD

**Issue by mattnelson** [corretto/corretto-docker#34]
There are alpine JRE tags for 8/11, but not for 15.

Can a 15 JRE image be added?

https://hub.docker.com/r/amazoncorretto/amazoncorretto/tags?page=1&name=jre

**Comment by luketn** [comment]
@mattnelson I hadn't come across this other set of docker images under /amazoncorretto/amazoncorretto:
https://hub.docker.com/r/amazoncorretto/amazoncorretto/tags?page=1&name=15

I guess these are community published images not official AWS ones?
![image](https://user-images.githubusercontent.com/1756555/95713174-d491c600-0cb1-11eb-8be9-94e00b051504.png)

The official images are under /amazoncorretto right?:
https://hub.docker.com/_/amazoncorretto

I raised another issue (#33) indicating that the documentation on the repository states that the official images support the 15 tags, but they are not actually published to DockerHub.

Let me know if I'm missing something and this new /amazoncorretto/amazoncorretto DockerHub will be where the official images are published for 15+.

**Comment by mattnelson** [comment]
I went off this comment for the new registry location.

https://github.com/corretto/corretto-docker/issues/15#issuecomment-672372087

**Comment by jguo11** [comment]
@mattnelson Our apology for the confusion. We had used the [amazoncorretto/amazoncorretto](https://hub.docker.com/r/amazoncorretto/amazoncorretto) repository for temporarily housing docker images that are not merged into official repository ([amazoncorretto](https://hub.docker.com/_/amazoncorretto)). However, in reality, there're a number of images co-exist in both repositories. We are planning to clean up `amazoncorretto/amazoncorretto` after the image are merged into the official repository by 11/20/2020. Please consider migrating to `amazoncorretto` repo if you can.

Regarding JRE images, we recommend building your own JRE images from the Dockerfile [11/alpine/Dockerfile](https://github.com/corretto/corretto-docker/blob/master/11/jre/alpine/Dockerfile) using [`jlink`](https://docs.oracle.com/javase/9/tools/jlink.htm#JSWOR-GUID-CECAC52B-CFEE-46CB-8166-F17A8E9280E9). Rather than providing a prebuild JRE image with fixed number of modules, we think it's more flexible for such you can choose only the modules you need.

**Comment by mattnelson** [comment]
> Regarding JRE images, we recommend building your own JRE images from the Dockerfile [11/alpine/Dockerfile](https://github.com/corretto/corretto-docker/blob/master/11/jre/alpine/Dockerfile?rgh-link-date=2020-10-14T00%3A44%3A30Z) using [`jlink`](https://docs.oracle.com/javase/9/tools/jlink.htm#JSWOR-GUID-CECAC52B-CFEE-46CB-8166-F17A8E9280E9). Rather than providing a prebuild JRE image with fixed number of modules, we think it's more flexible for such you can choose only the modules you need.

The benefit of the existing JREs images is that they are already over half the size of the JDK images. I imagine for some consumers the size tradeoff is acceptable to get an Amazon Corretto image off the shelf and avoid having to roll their own base layer. I do agree that the jlink guidance is sound for those consumers who desire a different set of linked modules.

**Comment by fl0ge** [comment]
@jguo11 I agree with @mattnelson about existing Alpine JRE images. They are much lighter than their Alpine JDK counterpart (44MB vs 186 MB for corretto 11) and in most cases, it is more than enough to be used as base image. Using custom Dockerfile using jlink can reduce a bit more the final size but it's not always worth it.
The initial issue is about JRE 15 but are you planning to update Alpine JRE 11 images with the latest Corretto 11.0.9 release? They are really useful and JRE 15 alpine will be also useful for us when we will be able to migrate from JDK/JRE 11.
Many thanks.

**Comment by luketn** [comment]
@jguo11 @mattnelson quick question: JRE was discontinued in Java 11:
"In this release, the JRE or Server JRE is no longer offered. Only the JDK is offered. Users can use jlink to create smaller custom runtimes."
https://www.oracle.com/java/technologies/javase/jdk-11-relnote.html#Important_Changes

What did you have in mind for the JRE 15 build? A specific set of modules in a paired-down JDK using jlink and or / some scripts to remove files which are rarely useful?

It's a genuine question because when I look at the modules in the JDK 15 I'm not sure what would be the minimal base set that would supply most applications with their requirements. 

```
➜  ~ docker run -it --entrypoint "/bin/sh" amazoncorretto:15-al2-jdk
sh-4.2# java --list-modules
java.base@15
java.compiler@15
java.datatransfer@15
java.desktop@15
java.instrument@15
java.logging@15
java.management@15
java.management.rmi@15
java.naming@15
java.net.http@15
java.prefs@15
java.rmi@15
java.scripting@15
java.se@15
java.security.jgss@15
java.security.sasl@15
java.smartcardio@15
java.sql@15
java.sql.rowset@15
java.transaction.xa@15
java.xml@15
java.xml.crypto@15
jdk.accessibility@15
jdk.aot@15
jdk.attach@15
jdk.charsets@15
jdk.compiler@15
jdk.crypto.cryptoki@15
jdk.crypto.ec@15
jdk.dynalink@15
jdk.editpad@15
jdk.hotspot.agent@15
jdk.httpserver@15
jdk.incubator.foreign@15
jdk.incubator.jpackage@15
jdk.internal.ed@15
jdk.internal.jvmstat@15
jdk.internal.le@15
jdk.internal.opt@15
jdk.internal.vm.ci@15
jdk.internal.vm.compiler@15
jdk.internal.vm.compiler.management@15
jdk.jartool@15
jdk.javadoc@15
jdk.jcmd@15
jdk.jconsole@15
jdk.jdeps@15
jdk.jdi@15
jdk.jdwp.agent@15
jdk.jfr@15
jdk.jlink@15
jdk.jshell@15
jdk.jsobject@15
jdk.jstatd@15
jdk.localedata@15
jdk.management@15
jdk.management.agent@15
jdk.management.jfr@15
jdk.naming.dns@15
jdk.naming.rmi@15
jdk.net@15
jdk.nio.mapmode@15
jdk.sctp@15
jdk.security.auth@15
jdk.security.jgss@15
jdk.unsupported@15
jdk.unsupported.desktop@15
jdk.xml.dom@15
jdk.zipfs@15
```

The compressed size of the AL2 15 docker image is 207.25 MB.
The actual JDK folder in the AL2 image is 279M.
```
sh-4.2# du -sh /usr/lib/jvm/java-15-amazon-corretto/
279M	/usr/lib/jvm/java-15-amazon-corretto/
```

When I look in the DockerHub repo I don't see any Corretto JRE 11 images:
https://hub.docker.com/_/amazoncorretto?tab=tags&page=1&name=11

Although in the GitHub repo there is a JRE build there which does a jlink to build a slimmer JDK:
```
jlink --endian little --release-info $JAVA_HOME/release \ 
        --add-modules "java.base,java.datatransfer,java.desktop,java.instrument,java.logging,java.management,java.management.rmi,java.naming,java.prefs,java.rmi,java.security.sasl,java.xml,jdk.internal.vm.ci,jdk.jfr,jdk.management,jdk.management.jfr,jdk.management.agent,jdk.net,jdk.sctp,jdk.unsupported,jdk.naming.rmi,java.compiler,jdk.aot,jdk.internal.vm.compiler,jdk.internal.vm.compiler.management,java.se,java.net.http,java.scripting,java.security.jgss,java.smartcardio,java.sql,java.sql.rowset,java.transaction.xa,java.xml.crypto,jdk.accessibility,jdk.charsets,jdk.crypto.cryptoki,jdk.crypto.ec,jdk.dynalink,jdk.httpserver,jdk.jsobject,jdk.localedata,jdk.naming.dns,jdk.scripting.nashorn,jdk.security.auth,jdk.security.jgss,jdk.xml.dom,jdk.zipfs,jdk.jdwp.agent,jdk.pack,jdk.scripting.nashorn.shell,jdk.jcmd,jdk.jfr" \
        --no-man-pages --no-header-files --strip-debug --output /temp/java-11-amazon-corretto
```

If you remove the modules which are no longer with us in 15 (jdk.scripting.nashorn, jdk.jdwp.agent, jdk.pack, jdk.scripting.nashorn.shell):
```
jlink --verbose \
      --add-modules "java.base,java.datatransfer,java.desktop,java.instrument,java.logging,java.management,java.management.rmi,java.naming,java.prefs,java.rmi,java.security.sasl,java.xml,jdk.internal.vm.ci,jdk.jfr,jdk.management,jdk.management.jfr,jdk.management.agent,jdk.net,jdk.sctp,jdk.unsupported,jdk.naming.rmi,java.compiler,jdk.aot,jdk.internal.vm.compiler,jdk.internal.vm.compiler.management,java.se,java.net.http,java.scripting,java.security.jgss,java.smartcardio,java.sql,java.sql.rowset,java.transaction.xa,java.xml.crypto,jdk.accessibility,jdk.charsets,jdk.crypto.cryptoki,jdk.crypto.ec,jdk.dynalink,jdk.httpserver,jdk.jsobject,jdk.localedata,jdk.naming.dns,jdk.security.auth,jdk.security.jgss,jdk.xml.dom,jdk.zipfs,jdk.jcmd,jdk.jfr" \
      --no-man-pages \
      --no-header-files \
      --strip-debug \
      --output ./java-minimal
```

The JDK comes down from 279MB to ~113MB.

I guess it might be difficult for AWS to manage the support burden of defining the 'JRE' as a specific subset of modules, or picking which modules to include?

**Comment by luketn** [comment]
Just linked a PR which could potentially address this issue? 
![image](https://user-images.githubusercontent.com/1756555/97066315-1f8bd180-1600-11eb-8b67-7b658bc06428.png)

This PR adds a 'slim' version of the JDK 15 docker files, which runs jlink to remove man-pages, header files and debug symbols whilst retaining all modules. 

The alpine image was 183MB smaller.
The Amazon Linux 2 image was 113MB smaller.

**Comment by mattnelson** [comment]
I believe this can be closed based on #38 and #50

**Comment by alvdavi** [comment]
I believe you're right
